marimo/_server/utils.py

```diff
@@ -44,12 +44,10 @@ def find_free_port(port: int, attempts: int = 100) -> int:
 
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
         try:
-            in_use = sock.connect_ex(("localhost", port)) == 0
-            if not in_use:
-                return port
+            sock.bind(("localhost", port))
+            return port
```

Comment:
Why this change?

Reply:
i reverted. it wasn't working as expected before, but this change made it not work even more